src/common/data-modeler-service/TableActions.ts

```diff
@@ -220,6 +222,39 @@ export class TableActions extends DataModelerActions {
     }
   }
 
+  @DataModelerActions.PersistentTableAction()
+  public async updateTableName(
+    { stateService }: PersistentTableStateActionArg,
+    tableId: string,
+    name: string
+  ): Promise<ActionResponse> {
+    const newName = sanitizeTableName(extractTableName(name));
+    const existingTable = stateService.getByField("tableName", newName);
+
+    if (existingTable) {
+      this.notificationService.notify({
+        message: `Another table with the name ${existingTable.tableName} already exists`,
```

Comment:
Should this be handled on the modal level? We can have some consistency by showing all validation errors in the modal dialog itself or having all as notification popups.

Reply:
Agreed. We should eventually move all notification to UI. But for now lets not add any new ones to backend.